cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -137,6 +137,24 @@ controller, and serviceaccounts controller.`,
 			}
 			cliflag.PrintFlags(cmd.Flags())
 
+			// as soon as we have the feature gates computed
+			// and before creating any clients or/and informers
+			// check to see if we need to enable WatchList for client-go
+			//
+			// setting the env var is required separately because
+			// we want to control components for which this feature
+			// is enabled
+			//
+			// see: https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/3157-watch-list
+			//
+			// note that we don't check if the env var was set when
+			// the FG wasn't enabled
+			if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.WatchList) {
```

Comment:
I was planning to promote the WatchList feature to `beta` in a separate PR. 
After that this FG will be `on` by default.